pkg/collectors/devices/fetcher.go

```diff
@@ -51,9 +51,44 @@ func (inst *fetcher) AddCommand(cmdInst *clients.Cmd) {
 	inst.cmdGrp.AddCommand(cmdInst)
 }
 
+func setValue(field *reflect.StructField, fieldVal reflect.Value, valueName string, value any) error {
+	//nolint:exhaustive //we could extend this but its not needed yet
+	switch field.Type.Kind() {
+	case reflect.String:
+
+		stringRes, ok := value.(string)
+		if !ok {
+			return fmt.Errorf("failed to convert %v into string", value)
+		}
+		fieldVal.SetString(stringRes)
+	case reflect.Slice:
+		resType := reflect.TypeOf(value)
+		if field.Type != resType {
+			return fmt.Errorf(
+				"type of %s (%v) does not match field type %v",
+				valueName,
+				resType,
+				field.Type,
+			)
+		}
+		if field.Type.Elem() != resType.Elem() {
+			return fmt.Errorf(
+				"InnerType of %s (%v) does not match field InnerType %v",
+				valueName,
+				resType.Elem(),
+				field.Type.Elem(),
+			)
+		}
+		fieldVal.Set(reflect.ValueOf(value))
+	default:
+		return fmt.Errorf("fetcher unmarshal not implemented for type: %s", field.Type.Name())
+	}
+	return nil
+}
+
 // unmarshall will populate the fields in `pack` with the values from `result` according to the fields`fetcherKey` tag.
 // fields with no `fetcherKey` tag will not be touched, and elements in `result` without a matched field will be ignored.
-func unmarshall(result map[string]string, pack interface{}) error {
+func unmarshall(result map[string]any, pack any) error {
```

Comment:
I think this would be more expressive as something like `unmarshall(data map[string]any, target any)`, or something similar. The current parameter names are not very descriptive.